### PR TITLE
[Eigen] new recipe

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

File: E/Eigen/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+cmake -DCMAKE_INSTALL_PREFIX=$prefix ..
+make -j${nproc}
+make install
+cd ..
+cp cmake/FindEigen3.cmake ${prefix}/share/eigen3/cmake/
+for lic in $(ls COPYING.*)  # different licences apply to different parts of Eigen
+do
+  install_license ${lic}
+done
```
Comment (giordano) [inline E/Eigen/build_tarballs.jl:26]: `install_license` does the loop already for you :slightly_smiling_face: 
```suggestion
install_license COPYING.*
```
Also, never use `ls` in this way: https://mywiki.wooledge.org/ParsingLs.  You could have just done
```sh
for lic in COPYING.*
```
:wink:

File: E/Eigen/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+# No products: Eigen is a pure header library
+products = Product[
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
```
Comment (giordano) [inline E/Eigen/build_tarballs.jl:39]: We don't it right now, but at some point in the future `dependencies` will need to have eltype `<: AbstractDependency` (similarly to `products`), so better enforce the type now
```suggestion
dependencies = Dependency[
```
:wink:

**Comment by dpo** [comment]
I removed the CMAKE_TOOLCHAIN stuff as it caused all builds to fail. How do I prevent it from trying to run an executable on Windows?

File: E/Eigen/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/eigen-eigen-323c052e1731
+
+mkdir build
+cd build/
+
+cmake -DCMAKE_INSTALL_PREFIX=$prefix ..
```
Comment (giordano) [inline E/Eigen/build_tarballs.jl:18]: Try to use the host toolchain
```suggestion
cmake -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} ..
```

**Comment by giordano** [comment]
> How do I prevent it from trying to run an executable on Windows?

Are you asking me why a package that would install only header files is trying to make sure the compiler works? :slightly_smiling_face:

File: E/Eigen/build_tarballs.jl (hunk 3 above)
Comment (dpo) [reply E/Eigen/build_tarballs.jl:18]: Tried that in the previous build. All builds failed.

Comment (dpo) [reply E/Eigen/build_tarballs.jl:18]: Ah my bad. It's not quite the same. Let me try.

**Pull request merged**